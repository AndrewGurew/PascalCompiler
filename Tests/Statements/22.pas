function foo(c,d:integer):integer;
begin
end
function bar(c:integer):integer;
begin
end
var
a,b:integer;
begin
a:=foo(a+b,bar(1)) + 1;
end