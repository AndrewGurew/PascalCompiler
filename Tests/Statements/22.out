Declaration
 ⎬b - Integer(Var)
 ⎬a - Integer(Var)
 ⎬bar - Function
 Params: c - Integer
 Return type: Integer



 ⎬foo - Function
 Params: d - Integer, c - Integer
 Return type: Integer



Main block
 ⎬ Assign
  ⎬a
  ⎬+
   ⎬foo with params:
    ⎬+
     ⎬a
     ⎬b
    ⎬bar with params:
     ⎬1
   ⎬1
End of Main block
